python/equistore/equistore/__init__.py

```diff
@@ -0,0 +1,39 @@
+# The equistore python package is distributed in multiple pieces, each with its
+# own version number and dependency management:
+#
+# - the `equistore` package contains only this file, and is the default
+#   user-facing package. It depends on the other packages, and re-export all
+#   classes/functions from them
+# - the `equistore-core` package contains the Python bindings to the
+#   equistore-core C API. Is is using Python's namespace packages
+#   (https://peps.python.org/pep-0420/) to install itself in `equistore/core`.
+#   The version of this package is the same as the version of the equistore-core
+#   rust crate. External package integrating with equistore such as rascaline
+#   only depend on equistore-core.
+# - the `equistore-operations` package contains the operations to manipulate
+#   equistore data.
+
+
+from . import core  # noqa
+
+try:
+    from . import operations  # noqa
+
+    HAS_EQUISTORE_OPERATIONS = True
+except ImportError:
+    HAS_EQUISTORE_OPERATIONS = False
```

Comment:
I thought equistore is a meta package only packaging the core and opearations together. Why aren't the only lines here:

```python
import equistore-core as core
import equistore-operations as operations

from equistore-core import TensorMap, TensorBlock, Labls
from equistore-operations import *
```

Reply:
Because equistore-core and equistore-operations are the names of the **distributions** in PEP language, and correspond to namespace **packages** installing their code in `<site-packages>/equistore/core` and `<site-packages>/equistore/operations`.

The weird imports is more about separating the import of sub-packages from the changes to the equistore namespace, but this could be changed.